src/internal/hooks/use-analytics/index.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect } from 'react';
+
+// We expect analytics is to be set only once and before the application is rendered.
+let analyticsEnabled: boolean | undefined = undefined;
+
+export function useAnalytics() {
```

Comment:
This is different than `useVisualRefresh` implementation: https://github.com/cloudscape-design/components/blob/main/src/internal/hooks/use-visual-mode/index.ts#L36

Should these two be unified to have the same activation requirements

Reply:
Comment can actually be removed. It started off similar but there's no expectation that this attribute should appear and apply to the entire page.

Reply:
Adapted based on feedback. Kept the comment as it now operates in a similar fashion as the `useVisualRefresh` hook.